test/e2e/event_test.go

```diff
@@ -116,6 +116,9 @@ func TestUpdateExistingEvent(t *testing.T) {
 		Type:   v1.EventTypeNormal,
 	}
 	event = framework.CreateEvent(t, event, event.InvolvedObject.Namespace)
+	// The exporter reconciles Events created during the same second as itself.
+	// Thus, to ensure that this Event is not reconciled, we sleep one second.
+	time.Sleep(time.Second)
```

Comment:
hmm is there no other solution for this but to sleep? What makes you time one second is enough and won't produce a flaky test? Can't we just remove the event from expected?

Reply:
In this case, I am certain that waiting for 1 second will make sure that this event will not be counted by the exporter.

When this event is reconciled by the exporter, its latest timestamp truncated to the second will be compared to the also truncated exporter running timestamp. Sleeping one second ensures that the event latest timestamp will always be before the exporter timestamp. Thus, making sure that this event will be filtered out.

The event is not expected here. If we don't sleep, it is sometimes counted and thus makes the test fail.

Reply:
Can you not do a wait poll and retry instead? That should give you a better result but less flaky IMO.

Reply:
The problem with that is I have nothing to wait for. Here I only need to increase my current time by one second so that the exporter will not count the Event. I agree that using sleep tends to make tests flaky, but AFAIK this applies mainly when we are waiting for something, which is not my case here.

Reply:
Not really, this is about retry and wait until you get the expected result so the event won't be counted, no?